cfpb/consumerfinance.gov
# P4C: optionally include parent plus in total debt

Include parent plus loans in total debt when checkbox is checked

---




## Additions

- Store `include parent plus` choice in URL
- Update `include parent plus` checkbox from URL on page load




## How to test this PR

Navigate to [grad path tool](localhost:8000/paying-for-college/your-financial-path-to-graduation/?iped=110635&pid=4510-3&houp=onCampus&typp=bachelors&lenp=4&ratp=inState&depp=dependent&houx=1395&fdx=498&clhx=127&trnx=750&hltx=265&entx=185&retx=282&taxx=143&othx=328). 
1. Verify that checking include box under Parent PLUS loans works correctly
    a. On `Make a plan` tab (under `Covering your costs`), open the `Parent PLUS loan` expandable and add a loan amount.
      <img width="603" alt="1" src="https://user-images.githubusercontent.com/778171/109033065-1f1d4900-767b-11eb-824d-0330944dab27.png">
    b. Navigate to `Is this too much debt?` (under `Affording your loans`) and verify that Parent PLUS loan amount is not included in the total debt.
      <img width="449" alt="2" src="https://user-images.githubusercontent.com/778171/109036780-cf408100-767e-11eb-85f4-989a2be541f0.png">

    c. Back on `Make a plan` tab, check the checkbox in the Parent PLUS loan section
      <img width="612" alt="3" src="https://user-images.githubusercontent.com/778171/109036804-d6678f00-767e-11eb-9abf-ad11310e8b5f.png">

    d. Now on the `Is this too much debt?`tab, Parent PLUS loan amount should be included in total debt
      
      <img width="606" alt="4" src="https://user-images.githubusercontent.com/778171/109036864-e8e1c880-767e-11eb-9fdf-a1c9c5cb5211.png">

2. Reload the page and verify that the Parent PLUS checkbox is still checked / loan amount is still included in total debt





## Checklist



  - PR titles are used to generate the change log in [releases](../../releases); good ones make that easier to scan.
  - Consider prefixing, e.g., "Mega Menu: fix layout bug", or "Docs: Update Docker installation instructions".
  - [This repo’s docs](https://cfpb.github.io/consumerfinance.gov/) (edit the files in the `/docs` folder) – for basic, close-to-the-code docs on working with this repo
  - CFGOV/platform wiki on GHE – for internal CFPB developer guidance
  - CFPB/hubcap wiki on GHE – for internal CFPB design and content guidance

### Front-end testing



#### Browser testing

Visually tested in the following supported browsers:



#### Accessibility


#### Other
